internal/repositories/github/service.go

```diff
@@ -325,6 +338,145 @@ func (r *repositoryService) DeleteByName(
 	return r.deleteRepository(ctx, client, &repo)
 }
 
+func (r *repositoryService) RefreshRepositoryByUpstreamID(
+	ctx context.Context,
+	upstreamRepoID int64,
```

Comment:
We might not need this method at all once we finish migrating, but it provides a lazy migration from the legacy tables.